src/Utils.js

```diff
@@ -80,7 +81,12 @@ export function nodeEqual(a, b) {
       return false;
     }
   }
-  return leftKeys.length === Object.keys(right).length;
+
+  if (typeof a !== 'string' && typeof a !== 'number') {
+    return leftKeys.length === Object.keys(right).length;
```

Comment:
It's part of ES5 - node 0.8 and higher, and IE 9 and higher have it.

However, if you want to support older engines (which I strongly recommend!) you can use the https://npmjs.com/object-keys module :-)